lib/rules/eol-last.js

```diff
@@ -39,7 +39,7 @@ module.exports = {
                 const sourceCode = context.getSourceCode(),
                     src = sourceCode.getText(),
                     location = {
-                        column: 1,
+                        column: lodash.last(sourceCode.lines).length,
```

Comment:
No, that all sounds great! I was thinking for the newline-is-present case
that there would be only two lines, the last ending in \n.

Can you do me a favor and double check for \r\n as well? Then I'll be happy
:-)

On Sep 13, 2016 4:38 PM, "kdex" <notifications@github.com> wrote:

> In lib/rules/eol-last.js
> <https://github.com/eslint/eslint/pull/7149#discussion_r78649544>:
>
> > @@ -39,7 +39,7 @@ module.exports = {
> >                  const sourceCode = context.getSourceCode(),
> >                      src = sourceCode.getText(),
> >                      location = {
> > -                        column: 1,
> > +                        column: lodash.last(sourceCode.lines).length,
>
> @platinumazure <https://github.com/platinumazure> I believe that the
> current implementation is correct and already uses zero-based columns.
> That's quite a statement, so I'll better back it up with a few examples to
> verify:
> Unwanted EOL
>
> /* eslint eol-last: [1, "never"] */\nvar a = 2;\n
>
> There are three lines in this example (verify by checking
> sourceCode.lines.length). The lengths of these lines are [35, 10, 0].
> The last line is where the error should be reported. location.column will
> be evaluated to "".length, which is 0. This sounds right for the column.
>
> After linting it, I could indeed verify that it uses column 0 (by logging
> it separately) and therefore, it reports:
>
> 3:1 warning Newline not allowed at end of file eol-last,
>
> as the output transforms indices to a 1-based format.
> Missing EOL
>
> Consider the following code:
>
> /* eslint eol-last: [1, "always"] */\nvar a = 2;
>
> The configuration implies that the code is faulty; we're missing an EOL
> character. This time, we only have two lines. The error should be reported
> in the last column of the line 2, which should be one index after ";" or in
> other words "var a = 2;".indexOf(";") + 1, which evaluates to 10.
>
> When linting, location.column will be evaluated to "var a = 2;".length,
> which evaluates to 10. Exactly what we wanted. Looking at the report, we
> can verify that it uses column 10 and therefore reports:
>
> 2:11 warning Newline required at end of file but not found eol-last
>
> as the output transforms indices to a 1-based format.
>
> Given that I haven't missed anything, this looks valid to me. 😄
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eslint/eslint/pull/7149/files/01c6b1eeb82950b94336b0a473312eb6ac3cdbb7#r78649544>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWessn2DF9Ol84IaGZxEbJvaLCs983ks5qpxfdgaJpZM4J8CdY>
> .
>